packages/server/src/service/csv.ts

```diff
@@ -79,3 +79,35 @@ export const toCSVSting = (infos: RichStateInfo[]) => {
   } as Record<StateKeys, any>))
   return toCSVStingGeneral(records, (simpleKeys as string[]).concat(compoundKeys))
 }
+
+class CSVMailingQueue {
+  /**
+   * Contains the uids of all users who have requested to download the csv
+   * file, so they don't trigger this action multiple times until it is finished.
+   */
+  private readonly uidDownloadingCsv: Set<string> = new Set()
```

Comment:
nit: uidDownloadingCSV (just because I find that most people like to keep acronyms all caps)